src/responder/server-nsq.ts

```diff
@@ -115,14 +125,34 @@ export class ServerNsq extends Server implements CustomTransportStrategy {
             topic,
             channel,
           });
+          let source;
           try {
-            await handler(packet.data, nsqCtx);
-            msg.finish();
+            source = await handler(packet.data, nsqCtx);
+            if (!isObservable(source)) {
+              return msg.finish();
+            }
           } catch (err) {
             this.logger.error(
-              `consumer reader failed to process message with error: ${err}, topic: ${topic}, channel: ${channel}`,
+              `consumer reader failed to process message with error: ${err.message}, topic: ${topic}, channel: ${channel}`,
             );
-            msg.requeue(this.options.requeueDelay || 90 * 1000);
+            msg.requeue(consumerOptions.requeueDelay || DEFAULT_REQUEUE_DELAY);
+          }
+
+          const wrappedErrObservable = source.pipe(
```

Comment:
Can this block be put in the try-catch section?

Reply:
I moved it out of try-catch on purpose, because here the `source` is an Observable, it has its own pattern to handle the errors, i.e. `catchError`. Putting it in the try-catch is meaningless.